pkg/dockerregistry/server/signaturedispatcher.go

```diff
@@ -0,0 +1,187 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+
+	ctxu "github.com/docker/distribution/context"
+
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/registry/api/errcode"
+	"github.com/docker/distribution/registry/api/v2"
+	"github.com/docker/distribution/registry/handlers"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+
+	gorillahandlers "github.com/gorilla/handlers"
+)
+
+const (
+	errGroup             = "registry.api.v2"
+	defaultSchemaVersion = 2
+)
+
+// signature represents a Docker image signature.
+type signature struct {
```

Comment:
It's strange to expose attributes of a private struct.

Also as a consumer of this API, I'd like to just use the struct to (de-)serialize json instead of writing my own.